modules/meshrenderinggl/src/processors/sphererasterizer.cpp

```diff
@@ -0,0 +1,261 @@
+           {BufferType::ColorAttrib, MeshShaderCache::Optional, "vec4"},
+           {BufferType::RadiiAttrib, MeshShaderCache::Optional, "float"},
+           {BufferType::ScalarMetaAttrib, MeshShaderCache::Optional, "float"}},
+
+          [&](Shader& shader) -> void {
+              shader.onReload([this]() { invalidate(InvalidationLevel::InvalidResources); });
+              configureShader(shader);
+          })} {
+
+    addPort(inport_);
+    addPort(outport_);
+
+    clipping_.addProperties(clipMode_, clipShadingFactor_, shadeClippedArea_);
+
+    sphereProperties_.addProperties(forceRadius_, defaultRadius_, forceColor_, defaultColor_,
+                                    useMetaColor_, metaColor_);
+
+    addProperties(renderMode_, forceOpaque_, useUniformAlpha_, uniformAlpha_, sphereProperties_,
+                  clipping_, transformSetting_, camera_, lighting_, trackball_);
+
+    clipShadingFactor_.readonlyDependsOn(
+        clipMode_, [](const auto& p) { return p == GlyphClippingMode::Discard; });
+    shadeClippedArea_.readonlyDependsOn(
+        clipMode_, [](const auto& p) { return p == GlyphClippingMode::Discard; });
+
+    transformSetting_.setCollapsed(true);
+    camera_.setCollapsed(true);
+    lighting_.setCollapsed(true);
+    trackball_.setCollapsed(true);
+}
+
+void SphereRasterizer::process() {
+    std::shared_ptr<const Rasterization> rasterization =
+        std::make_shared<SphereRasterization>(*this);
+
+    auto data = (transformSetting_.transforms_.empty())
+                    ? rasterization
+                    : std::make_shared<TransformedRasterization>(rasterization,
+                                                                 transformSetting_.getMatrix());
+    outport_.setData(data);
+}
+
+void SphereRasterizer::initializeResources() {
+    for (auto& item : shaders_->getShaders()) {
+        configureShader(item.second);
+    }
+}
+
+void SphereRasterizer::configureShader(Shader& shader) {
+    utilgl::addDefines(shader, lighting_);
+    shader[ShaderType::Vertex]->setShaderDefine("FORCE_RADIUS", forceRadius_);
+    shader[ShaderType::Vertex]->setShaderDefine("FORCE_COLOR", forceColor_);
+    shader[ShaderType::Vertex]->setShaderDefine("USE_SCALARMETACOLOR", useMetaColor_);
+    shader[ShaderType::Fragment]->setShaderDefine("SHADE_CLIPPED_AREA", shadeClippedArea_);
+    shader[ShaderType::Fragment]->setShaderDefine("DISCARD_CLIPPED_GLYPHS",
+                                                  clipMode_.get() == GlyphClippingMode::Discard);
+    shader[ShaderType::Geometry]->setShaderDefine("DISCARD_CLIPPED_GLYPHS",
+                                                  clipMode_.get() == GlyphClippingMode::Discard);
+
+    configureOITShader(shader);
+    shader.build();
+}
+
+void SphereRasterizer::configureOITShader(Shader& shader) {
+    auto fso = shader.getFragmentShaderObject();
+
+    fso->addShaderExtension("GL_NV_gpu_shader5", true);
```

Comment:
Lots of extensions here, what
If they are not supported what if it's not on nvidia? Should one not test if the extension is available first?

Reply:
As far as I understood it, the rasterizer has an internal state whether the A-buffer stuff is supported.

Reply:
But It's your code that's setting the extensions, that does not make any sense

Reply:
should be addressed now